tools/automatic_query_fixer/src/main/java/com/google/cloud/bigquery/utils/queryfixer/fixer/TableNotFoundFixer.java

```diff
@@ -100,24 +107,43 @@ private String constructFullTableName(String projectId, String datasetId, String
     return String.format("%s.%s.%s", projectId, datasetId, tableName);
   }
 
-  private int findTheIndexOfIncorrectTable() {
-    // The table in the error message is presented in the legacySQL mode, but this fixer is used to
-    // fix the
-    // standardSQL. Thus, the table name needs to be converted to the one consistent with
-    // standardSQL.
-    // The change is from project:dataset.table to project.dataset.table.
-    String tableName = err.getTableName().replace(':', '.');
-    int index = query.indexOf(tableName);
-
-    // Since the TableNotFound error has no position info, this method will convert the index to the
-    // position and assign to the `err`.
-    Position position = queryPositionConverter.indexToPos(index);
-    this.err.setErrorPosition(position);
-    return index;
+  private SubstringView findSubstringViewOfIncorrectTable(TableId fullTableId) {
+    String regex;
+    // If the project ID is the default one, then the actual table may not include project ID,
+    // which looks like "dataset.table".
```

Comment:
Theoretically I think we should handle single 'table' too? We can add TODO here.

Reply:
Yes, if a user specifies the default dataset in the input options. TODO is added.